libcontainerd/client_linux.go

```diff
@@ -274,18 +267,6 @@ func (clnt *client) setExited(containerID string) error {
 			ExitCode: exitCode,
 		}})
 
-	// Unmount and delete the bundle folder
```

Comment:
You are missing the removal of `unmount` in 'clean()`. Maybe we would need to keep this unmount code here in some form though in case there was a ungraceful shutdown between the restarts. This method is only called on restore and could handle that case.